include/pros/imu.h

```diff
@@ -0,0 +1,295 @@
+/**
+ * \file pros/imu.h
+ *
+ * Contains prototypes for functions related to the VEX Inertial sensor.
+ *
+ * Visit https://pros.cs.purdue.edu/v5/tutorials/topical/imu.html to learn
+ * more.
+ *
+ * This file should not be modified by users, since it gets replaced whenever
+ * a kernel upgrade occurs.
+ *
+ * Copyright (c) 2017-2019, Purdue University ACM SIGBots.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#ifndef _PROS_IMU_H_
+#define _PROS_IMU_H_
+
+#include <stdbool.h>
+#include <stdint.h>
+
+#ifdef __cplusplus
+extern "C" {
+namespace pros {
+namespace c {
+#endif
+
+typedef enum imu_status_e {
+	E_IMU_STATUS_CALIBRATING = 0x01,
+	E_IMU_STATUS_ERROR = 0xFF,  // NOTE: used for returning an error from the get_status function, not that the IMU is
+	                            // necessarily in an error state
+} imu_status_e_t;
+
+typedef struct __attribute__((__packed__)) quaternion_s {
+	double x;
+	double y;
+	double z;
+	double w;
+} quaternion_s_t;
+
+struct __attribute__((__packed__)) imu_raw_s {
+	double x;
+	double y;
+	double z;
+};
+
+typedef struct imu_raw_s imu_gyro_s_t;
+typedef struct imu_raw_s imu_accel_s_t;
+
+typedef struct __attribute__((__packed__)) euler_s {
+	double pitch;
+	double roll;
+	double yaw;
+} euler_s_t;
+
+/**
+ * Calibrate IMU
+ *
+ * This takes approximately 2 seconds, and is a non-blocking operation.
+ *
+ * This function uses the following values of errno when an error state is
+ * reached:
+ * ENXIO - The given value is not within the range of V5 ports (1-21).
+ * ENODEV - The port cannot be configured as an Inertial Sensor
+ * EAGAIN - The sensor is already calibrating
+ *
+ * \param port
+ *        The V5 Inertial Sensor port number from 1-21
+ * \return 1 if the operation was successful or PROS_ERR if the operation
+ * failed, setting errno.
+ */
+int32_t imu_reset(uint8_t port);
+
+/**
+ * Get the total number of degrees the Inertial Sensor has spun about the z-axis
+ *
+ * This value is theoretically unbounded. Clockwise rotations are represented
+ * with positive degree values, while counterclockwise rotations are represented
+ * with negative ones.
+ *
+ * This function uses the following values of errno when an error state is
+ * reached:
+ * ENXIO - The given value is not within the range of V5 ports (1-21).
+ * ENODEV - The port cannot be configured as an Inertial Sensor
+ * EAGAIN - The sensor is still calibrating
+ *
+ * \param  port
+ * 				 The V5 Inertial Sensor port number from 1-21
+ * \return The degree value or PROS_ERR_F if the operation failed, setting
+ * errno.
+ */
+double imu_get_rotation(uint8_t port);

```

Comment:
imu_get_rotations sounds better to me, since it's unbounded.

Reply:
"Rotations" implies a quantity of full revolutions to me. I'm still not a fan of "rotation" either as it doesn't specify an axis, but I don't have anything better to offer.

Reply:
gonna stick with imu_get_rotation